### PR TITLE
py-pygithub: new versions, dependencies

### PR DESCRIPTION
**Pull request by wdconinc**
This adds the most recent v1 version, and the latest version (v2). Updated dependencies (from requirements.txt). The `py-pyjwt@2.4.0:` requirement was to avoid a CVE in pyjwt, not for a technical reason.

### CODE REVIEW

**Review by tldahlgren — COMMENTED** [review]
FYI. Confirmed the two new version sha256.

**Review by manuelakuhn — DISMISSED** [review]
Optional:
It makes reviewing easier if you specify the dependencies in the same order as they are listed in the package requirements (here of the newest package)

Other than that, LGTM

**Comment by wdconinc** [comment]
> in the same order as they are listed in the package requirements

Reordered. Thanks for the suggestion.

**Review by manuelakuhn — DISMISSED** [review]

File: var/spack/repos/builtin/packages/py-pygithub/package.py [hunk 1]
```diff
@@ -7,16 +7,24 @@
...
     homepage = "https://pygithub.readthedocs.io/"
     pypi = "PyGithub/PyGithub-1.54.1.tar.gz"
 
+    version("2.1.1", sha256="ecf12c2809c44147bce63b047b3d2e9dac8a41b63e90fcb263c703f64936b97c")
+    version("1.59.1", sha256="c44e3a121c15bf9d3a5cc98d94c9a047a5132a9b01d22264627f58ade9ddc217")
     version("1.55", sha256="1bbfff9372047ff3f21d5cd8e07720f3dbfdaf6462fcaed9d815f528f1ba7283")
 
     depends_on("python@3.6:", type=("build", "run"))
+    depends_on("python@3.7:", type=("build", "run"), when="@1.57:")
+
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-pygithub/package.py:21]: Missing build dep on py-setuptools-scm

**Review by adamjstewart — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/py-pygithub/package.py [hunk 2]
```diff
@@ -7,16 +7,24 @@
...
     depends_on("py-setuptools", type="build")
```
Comment (wdconinc) [inline var/spack/repos/builtin/packages/py-pygithub/package.py:22]:
```suggestion
    depends_on("py-setuptools", type="build")
    depends_on("py-setuptools-scm", type="build", when="@1.58.1:")
```

File: var/spack/repos/builtin/packages/py-pygithub/package.py (hunk 1 above)
Comment (wdconinc) [reply var/spack/repos/builtin/packages/py-pygithub/package.py:21]: In 4a806ef.

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**